ouroboros-network/test/Test/Ouroboros/Network/KeepAlive.hs

```diff
@@ -149,99 +147,56 @@ prop_keepAlive_convergenceM
         , MonadTimer m
         , MonadThrow (STM m)
         )
-    => NetworkDelay
+    => Tracer m (TraceKeepAliveClient String)
+    -> NetworkDelay
     -> Int
-    -> m Property
-prop_keepAlive_convergenceM (NetworkDelay nd) seed = do
+    -> m ()
+prop_keepAlive_convergenceM tracer (NetworkDelay nd) seed = do
     registry <- newFetchClientRegistry
     controlMessageV <- newTVarM Continue
     let controlMessageSTM = readTVar controlMessageV
         clientId = "client"
         timeConstant = 1000 -- Same as in PeerGSV's <> definition
         keepAliveInterval = 10
 
-    (c_aid, s_aid) <- runKeepAliveClientAndServer (NetworkDelay nd) seed verboseTracer controlMessageSTM
+    (c_aid, s_aid) <- runKeepAliveClientAndServer (NetworkDelay nd) seed tracer controlMessageSTM
                           registry clientId (KeepAliveInterval keepAliveInterval)
     threadDelay $ timeConstant * keepAliveInterval
-    dqLive <- atomically $ readPeerGSVs registry
 
     atomically $ writeTVar controlMessageV Terminate
     void $ wait c_aid
     void $ wait s_aid
 
     -- XXX Must be larger than the KeepAliveInterval timeout or we leak threads in the SIM
     threadDelay (keepAliveInterval + 128)
```

Comment:
I think you only need to use this because `runPeerWithLimits` will leak a thread (on Windows and `IOSim`).  If this is the case #2631 will fix it.  It's worth adding this as a comment.

Reply:
I'll add a reverence to it.